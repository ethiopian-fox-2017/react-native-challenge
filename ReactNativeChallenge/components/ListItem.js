import React from 'react';
import {
  View,
  ScrollView,
  Text
} from 'react-native';

import Item from './Item';

const ListItem = (props) => {
  return (
    <View>
      <ScrollView>
    {props.items.length === 0 && <Text>Loading...</Text>}

      { props.items.map((item, index) => {
        // console.log(item);
        return (
        <Item key={index} itemName={item.name} itemId={index + 1} ></Item>
        )

      })
      }
      </ScrollView>
    </View>
  )
}

export default ListItem;
