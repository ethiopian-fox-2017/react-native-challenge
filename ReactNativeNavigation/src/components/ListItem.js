import React from 'react';
import { connect } from 'react-redux';
import {
  View,
  ScrollView,
  Text,
  StatusBar
} from 'react-native';

import Item from './Item';
import { fetchPhoto } from '../actions';

class ListItem extends React.Component {

  constructor(props){
    super(props);
  }

  static navigationOptions = { title: 'Hello Photographer', };

  componentDidMount(){
    this.props.fetchPhoto();
  }

  render() {
    return (
      <View style={styles.container}>

      <StatusBar backgroundColor="#3F51B5"
        barStyle="light-content" />

        <Text style={styles.welcome}>
          Photography
        </Text>
        <ScrollView >
        <View style={styles.viewStyle}>
      {this.props.photos.length === 0 && <Text>Loading...</Text>}

        { this.props.photos.map(item => {
          // console.log(item);
          return (
          <Item key={item.id} item={item}
            navigation = {this.props.navigation}
          />
          )

        })
        }
        </View>
        </ScrollView>
      </View>
    )
  }
}

const styles = {
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    marginTop: 20,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
  viewStyle: {
    flexDirection: 'row',
    flexWrap: 'wrap'
  }
};

const mapStateToProps = (state) => {
  return {
    photos: state.photos
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchPhoto: () => dispatch(fetchPhoto())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ListItem);
