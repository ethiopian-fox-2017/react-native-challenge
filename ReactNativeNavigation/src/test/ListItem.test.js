import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';

import ListItem from '../components/ListItem';
import store from '../store';


it('render and match to snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <ListItem />
    </Provider>
  )
  expect(tree).toMatchSnapshot();
})
