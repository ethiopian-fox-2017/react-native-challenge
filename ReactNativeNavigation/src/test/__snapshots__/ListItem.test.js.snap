// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render and match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    Photography
  </Text>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Loading...
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
